docs/cookbook/opentelemetry.md

```diff
@@ -0,0 +1,297 @@
+
+## 3. Configure OpenTelemetry
+
+There are various ways to configure and integrate OpenTelemetry in a project. We think this is the cleanest way.
+
+Create `telemetry.js` in the project root:
+
+```js
+// ./telemetry.js
+const { NodeSDK, resources } = require('@opentelemetry/sdk-node');
+const { SemanticResourceAttributes } = require('@opentelemetry/semantic-conventions');
+const { getNodeAutoInstrumentations } = require('@opentelemetry/auto-instrumentations-node');
+const { JaegerExporter } = require('@opentelemetry/exporter-jaeger');
+
+// 1. Add the application metadata (resource)
+const pkg = require('./package-lock.json');
+const resource = new resources.Resource({
+  [SemanticResourceAttributes.SERVICE_NAME]: pkg.name,
+  [SemanticResourceAttributes.SERVICE_VERSION]: pkg.version
+});
+
+// 2. Initialize the exporter
+const traceExporter = new JaegerExporter({
+  tags: [],
+  endpoint: 'http://localhost:14268/api/traces'
+});
+
+// 3. Initialize the SDK
+const sdk = new NodeSDK({
+  resource,
+  traceExporter,
+  instrumentations: [ getNodeAutoInstrumentations() ]
+});
+
+// 4. The shutdown handler
+const shutdown = async () => {
+  await sdk
+    .shutdown()
+    .then(
+      () => console.log('OpenTelemetry stopped'),
+      (err) => console.log('Error shutting down OpenTelemetry', err)
+    );
+};
+
+module.exports = {
+  sdk,
+  shutdown
+};
+
+```
+
+Now we'll need to refactor `app.js` a little bit to connect OpenTelemetry with Apostrophe:
+
+```js
+// ./app.js
+// The Apostrophe bootstrap
+const apostrophe = require('apostrophe');
+const { sdk, shutdown } = require('./telemetry');
+
+// Move Apostrophe configuration to a variable, rather than directly
+// invoking the apostrophe function
+
+const config = {
+  shortName: 'myApp',
+  // Provide our shutdown handler
+  beforeExit: shutdown,
+  modules: {
+    // ... Our regular module configuration comes here
+  }
+};
+
+// Invoke based on APOS_OPENTELEMETRY environment variable
+if (process.env.APOS_OPENTELEMETRY) {
+  // This allows the SDK to be fully initialized,
+  // and is the cleanest way to start the app
+  sdk
+    .start()
+    .then(() => console.log('OpenTelemetry started'))
+    .then(() => apostrophe(config));
+} else {
+  // Start normally if APOS_OPENTELEMETRY is not truthy
+  apostrophe(config);
+}
+```
+
+The important bits of this change are:
+
+1. Capture the apostrophe configuration in a variable
+2. Provide a `shutdown` handler for `telemetry.js` as the value of the Apostrophe `beforeExit` configuration option 
+3. Start Apostrophe after OpenTelemetry has been started
+
+## 4. Start the application
+
+Be sure that the Jaeger backend is running (step 2) and open http://localhost:16686 in the browser. 
+
+Start the app:
+
+```sh
+APOS_OPENTELEMETRY=1 node app
+```
+
+After Apostrophe is fully started, refresh the Jaeger UI and you should see some data shown (give it some time).
+
```

Comment:
it might mention `apostrophe:boot` operation name here, just to be consistent with the instruction below that has a tip what the root operation is.